File: src/JuegoDeRol/versionJava/Mago.java
package JuegoDeRol.versionJava;

import java.util.ArrayList;
import java.util.List;

public class Mago extends Personaje {
    private List<Hechizo> hechizos;

    public Mago(String nombre, int hp, int dmge) {
        super(nombre, hp, dmge);
        this.hechizos = new ArrayList<>();
    }

    public void aprenderHechizo(Hechizo hechizo) {
        hechizos.add(hechizo);
        System.out.println(getNombre() + " ha aprendido " + hechizo.getNombre());
    }

    public void usarHechizo(Personaje enemigo, Hechizo hechizo) {
        System.out.println(getNombre() + " ha usado " + hechizo.getNombre() + " contra " + enemigo.getNombre());
        recibirDamage(hechizo.getDmage(), enemigo);
    }
}

File: src/JuegoDeRol/versionKotlin/Mago.kt
package org.example.JuegoDeRol.versionKotlin

class Mago(nombre:String, hp: Int, dmge: Int):Personaje(nombre,hp,dmge) {

    val hechizos = mutableListOf<Hechizo>()


    fun aprenderHechizo(hechizo: Hechizo){
        hechizos.add(hechizo)
        println("$nombre ha aprendido ${hechizo.nombre}")
    }

    fun usarHechizo(enemigo: Personaje, hechizo: Hechizo){

        println("$nombre ha usado ${hechizo.nombre} contra ${enemigo.nombre}")

        recibirDamage(hechizo.dmage, enemigo)
    }


}